File: 2453-destroy-sequential-targets/2453-destroy-sequential-targets.java
class Solution {
  public int destroyTargets(int[] A, int space) {
        int res = (int)1e9;
        HashMap<Integer, Integer> count = new HashMap<>();
        for (int a : A)
            count.put(a % space, count.getOrDefault(a % space, 0) + 1);
        int maxc = Collections.max(count.values());
        for (int a : A)
            if (count.get(a % space) == maxc)
                res = Math.min(res, a);
        return res;
    }
}

File: 2453-destroy-sequential-targets/2453-destroy-sequential-targets.cpp
class Solution {
public:
  int destroyTargets(vector<int>& A, int space) {
        int res = 1e9, maxc = 0;
        unordered_map<int, int> count;
        for (int a : A)
            maxc = max(maxc, ++count[a % space]);
        for (int a : A)
            if (count[a % space] == maxc)
                res = min(res, a);
        return res;
    }
};